### DISCUSSION TITLE
Support Python version 3.8+ and keep dremio-arrow in sync with Python security release versions.

### DISCUSSION THREAD

**Discussion by rxm7706** [jaysnm/dremio-arrow#9] [Ideas]
https://devguide.python.org/versions/

Supporting only Python 3.10+ might be a little too progressive and end up isolating some users.
Ideally we would want any user of an active version of python - to be a customer and user of dremio-arrow.
Can we considering supporting Python 3.8 + ? 

![image](https://github.com/jaysnm/dremio-arrow/assets/95496360/093aa942-ba02-4381-973e-fcc30fe94b3d)

@jaysnm @elanqo please opine.

**Comment by jaysnm** [comment]
The idea for supporting python3.10 and above is to take advantage of advancements in type hints and other niceties built-in into python3.10 up. However, it is possible to roll back support to python 3.8 as the upgrade didn't involve any breaking change.
